### ISSUE TITLE
cannot connect correctly with etcd on localhost

### ISSUE THREAD

**Issue by jonniedarko** [stianeikeland/node-etcd#40]
my etcd instance is running locally and I can perform get and set using the cli
```
// set
etcdctl set /services/myservice "{\"hostname\": \"127.0.0.1\", \"port\": 3000}"

//get 
etcdctl get services/myservice 
// results in -> {"hostname": "127.0.0.1", "port": 3000}
```

but when I use your module

```js
var Etcd = require('node-etcd');
var etcd = new Etcd(['http://localhost:2379', 'http://localhost:4001']);
etc.get('services/myservice', { wait: true}, function (err, value){
    console.log(err, value);
});
```
Can you explain why my value is HTML (see below)
I also find that when I don't include hosts and just use `new Etcd()` I get an internal HTML page with references to proxy issues: **"The requested website is blocked by [insert company name] Proxy due to potential malicious activity or other security reasons."**

```html
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" data-adblockkey="MFwwDQYJKoZIhvcNAQEBBQADSwAwSAJBAKrfIMFkSaoTSqKmC+BrghK0CpDHc0MuVzmMHin8LIORhpXbped+iYhSnZurWnEO0zcKcVIrzp026LVc5pMB9bUCAwEAAQ==_mHzvOc3GBNo9sxVki+5Fe+E9GsbiT2+mBIEtU+yAIW6jw6Uds6VUD5RSkqMfuUSrcw4AYCrFOWaaWSRGDzU8pw==">

<head>
    <script type="text/javascript">
        var abp;
    </script>
    <script type="text/javascript" src="http://www.http.com/px.js?ch=1"></script>
    <script type="text/javascript" src="http://www.http.com/px.js?ch=2"></script>
    <script type="text/javascript">
        function handleABPDetect() {
            try {
                var imglog = document.createElement("img");
                imglog.src = "http://www.http.com/rg-logabpstatus.php?a=WVpqOEdZZldDTWo4VWF0M0pNa1pBek0xQnE1WVlqR2VaQ0Frd0V4TUxmcDAvOFN2WnhHYS8weFFEeTZTRHlleTM0UFk3UDhzemxqSWFRdXlFdXpIUGc9PQ==&b=" + abp;
                document.body.appendChild(imglog);
            } catch (err) {}
        }
        try {
            window.onload = handleABPDetect();
        } catch (err) {}
    </script>
    <title>http.com</title>
    <meta http-equiv="Content-Type" content="text/html; charset=">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable = no">
    <style type="text/css">
        * {
            margin: 0;
            padding: 0
        }
        
        a {
            text-decoration: none;
            outline: none
        }
        
        a:hover {
            text-indent: 0;
            cursor: pointer
        }
        
        .clearfix:after {
            visibility: hidden;
            display: block;
            font-size: 0;
            content: " ";
            clear: both;
            height: 0
        }
        
        * html .clearfix {
            zoom: 1
        }
        
        *:first-child+html .clearfix {
            zoom: 1
        }
        
        body {
            font-family: Arial, Helvetica, sans-serif;
            background: url(http://c.rmgserving.com/rmgpsc/7867/body-bg.gif) 0 0 repeat #000
        }
        /*.arrows{background:url(http://b.rmgserving.com/rmgpsc/7867/arrows.jpg)}*/
        
        #main-wrap {
            background: url(http://a.rmgserving.com/rmgpsc/7867/header-bg.jpg) top center no-repeat;
            background-size: 100% 100px
        }
        
        .container {
            width: 960px;
            margin: 0 auto
        }
        
        .header {
            height: 100px
        }
        /*commented for new design */
        /*.leftblk{float:left; overflow:hidden}
.leftblk img{float: left; margin-top: 23px; padding-right: 15px;}*/
        /*.domain_name{float:left; line-height:100px; font-size:26px; font-weight:normal; color:#fff}*/
        
        .searchbox {
            float: right;
            width: 410px;
            * width: 420px;
            _width: 420px;
            height: 28px !important;
            padding-top: 2px;
            margin-top: 35px;
            /*padding-left:33px;background:url(http://d.rmgserving.com/rmgpsc/7868/srch-bg-caf2.gif) 0 0 no-repeat*/
        }
        
        .content {
            padding-bottom: 10px;
            background: #202020
        }
        /**RESPOSIVE CHANGES MADE HERE**/
        
        .domain_name {
            line-height: normal;
            font-size: 26px;
            font-weight: normal;
            color: #fff;
            margin-top: 35px
        }
        
        .leftblk {
            float: left;
            width: 39%;
            word-wrap: break-word;
            background: url(http://c.rmgserving.com/rmgpsc/7867/logo1.png) no-repeat center left;
            padding-left: 70px;
            height: 100px;
            overflow: hidden;
        }
        
        .frt_arr {
            float: left;
            width: 30%;
            height: 425px;
            background: url(http://d.rmgserving.com/rmgpsc/7983/frst_arr.jpg);
            background-position: 10px 70px;
            background-repeat: no-repeat;
            background-size: 90%
        }
        
        .lst_arr {
            float: left;
            width: 30%;
            height: 425px;
            background: url(http://b.rmgserving.com/rmgpsc/7983/last_arr.jpg);
            background-position: 0px 70px;
            background-repeat: no-repeat;
            background-size: 90%
        }
        
        .kwd_bloack {
            float: left;
            width: 40%;
            margin-top: 74px
        }
        /**RESPOSIVE CHANGES END HERE**/
        
        .bottom_bg {
            height: 44px;
            /*background:#181818; */
            margin-top: 34px
        }
        
        .bottom_rs {
            width: 960px;
            height: 44px;
            overflow: hidden;
            margin: 0 auto;
            position: relative
        }
        
        #bot_rs {
            padding-top: 9px
        }
        
        .separator {
            position: absolute;
            top: 0;
            left: 156px;
            width: 5px;
            height: 44px;
            border-left: 2px solid #282828;
            background: url(http://a.rmgserving.com/rmgpsc/7867/sep-arw.gif) center center no-repeat
        }
        
        .footer-nav {
            width: 100%;
            height: 74px;
            text-align: center;
            color: #c0c0c0;
        }
        
        .footer-nav a {
            font-size: 12px;
            line-height: 74px;
            color: #c0c0c0;
            padding: 0 5px;
            text-decoration: underline
        }
        
        .footer-nav a:hover {
            text-decoration: none
        }
        
        .inquire {
            text-align: right;
            padding-top: 10px;
            color: #fff
        }
        
        .inquire a {
            font-size: 12px;
            font-weight: normal;
            color: #fff
        }
        
        .sale-msg {
            background: #fff;
            color: #4b4b4b;
            text-align: center;
            font-size: 14px;
            width: 100%;
            top: 0;
            left: 0
        }
        
        .sale-msg a {
            text-decoration: none;
            color: #079ce9;
            font-size: 14px;
        }
        
        .sale-msg a:hover,
        .bottom_rs ul li a:hover,
        .mid_rs ul li a:hover,
        .inquire a:hover {
            text-decoration: underline
        }
        
        @media not all and (min-width: 970px) {
            .container {
                width: 95%
            }
            .bottom_bg {
                display: none
            }
            .footer-nav a {
                line-height: 45px
            }
        }
        
        @media not all and (min-width: 840px) {
            .frt_arr {
                display: none
            }
            .lst_arr {
                display: none
            }
            .kwd_bloack {
                float: none;
                width: 95%;
                margin: 0 auto;
                padding-top: 10px
            }
            .leftblk {
                float: none;
                height: 50px;
                background-size: 35px;
                padding-left: 50px;
                width: auto
            }
            .domain_name {
                line-height: normal;
                font-size: 20px;
                margin-top: 0;
                padding-top: 15px
            }
            .searchbox {
                float: none;
                margin-top: 10px;
                padding-top: 0;
                width: auto;
            }
        }
    </style>
    <script type="text/javascript" src="http://www.google.com/adsense/domains/caf.js"></script>
</head>

<body onload="" onunload="" onBeforeUnload="">
    <noscript>
        <meta http-equiv="refresh" content="0;url=http://www.http.com/rg-erdr.php?_rpo=t ntZzFFI&_rdm=9f1NJXf43BJvmfNf4.JvV&p=J1c4gcvmfN9ffFGoFgXoYVa%7C%40%7CbzGzTE%2C5f95zbT%2C5f95Et%7C%40%7CX55g.JvV%7C%40%7C%7C%40%7C%7C%40%7CZzbHzEZHE%7C%40%7CzHZZFTH%7C%40%7CK-%7C%40%7C19c4NW4cGFHzTEtTEGZGzTbT%7C%40%7Ct+nh8llbj%7C%40%7Ct+7zFZKFH&ga=5rKI1sSSBwFmmNY4RDXg%2BB7VS0yzNgnQ9JpHpsLHSsvFu0NyshpFxWu0exObFAo5IdFZQrTZujcZjrXONxp3PxtPV7ezER14zh7kWG69e5lf08niAgSfUKiyBSByBIQgw92X314XjQy5h09a65wi7JIhzrLV0PReUb%2F7SMvquCNLeg70qvVfr7kkYTx9Z4NO&t=gnojs" />
        <center>
            <p style="padding:1em; font-size:1.5em;">For search results please <a href="http://www.http.com/rg-erdr.php?_rpo=t ntZzFFI&_rdm=9f1NJXf43BJvmfNf4.JvV&p=J1c4gcvmfN9ffFGoFgXoYVa%7C%40%7CbzGzTE%2C5f95zbT%2C5f95Et%7C%40%7CX55g.JvV%7C%40%7C%7C%40%7C%7C%40%7CZzbHzEZHE%7C%40%7CzHZZFTH%7C%40%7CK-%7C%40%7C19c4NW4cGFHzTEtTEGZGzTbT%7C%40%7Ct+nh8llbj%7C%40%7Ct+7zFZKFH&ga=5rKI1sSSBwFmmNY4RDXg%2BB7VS0yzNgnQ9JpHpsLHSsvFu0NyshpFxWu0exObFAo5IdFZQrTZujcZjrXONxp3PxtPV7ezER14zh7kWG69e5lf08niAgSfUKiyBSByBIQgw92X314XjQy5h09a65wi7JIhzrLV0PReUb%2F7SMvquCNLeg70qvVfr7kkYTx9Z4NO&t=gnojs" style="text-decoration:underline; color:#0000EE;">CLICK HERE</a>.</p>
        </center>
    </noscript>
    <script type="text/javascript" language="javascript">
        function __gpup(url, height, width, name) {
            sw = window.open(url, name, 'height=' + height + ',width=' + width + ',location=no,toolbar=0,resizable=1,scrollbars=1');
            if (window.focus) {
                sw.focus()
            };
            return false;
        };
    </script>
    <script text="text/javascript">
        var design = {
            'pageOptions': {
                'pubId': '',
                'resultsPageBaseUrl': '',
                'fontFamily': 'arial',
                'hl': 'en',
                'maxTermLength': 29,
                'adtest': 'on',
                'clicktrackUrl': '',
                'fontFamilyAttribution': 'arial',
                'type': 'pageoptions',
                'pageLoadedCallback': function(requestAccepted, status) {
                    document.body.style.visibility = 'visible';
                    if (!requestAccepted) {}
                }
            },
            'searchboxBlock': {
                'container': 'searchbox',
                'type': 'searchbox',
                'fontSizeSearchInput': 16,
                'fontSizeSearchButton': 13,
                'widthSearchInput': 215,
                'colorSearchButton': '#e2e2e2',
                'hideSearchInputBorder': false,
                'hideSearchButtonBorder': false
            },

            'mainrs': {
                'container': 'main_rs',
                'type': 'relatedsearch',
                'colorBackground': 'transparent',
                'number': 6,
                'fontSizeTitle': 19,
                'titleBold': true,
                'colorTitleLink': '#079ce9',
                'noTitleUnderline': false,
                'lineHeightTitle': 30,
                'fontFamily': 'arial',
                'horizontalAlignment': 'center',
                'fontSizeAttribution': 13,
                'colorAttribution': '#ababab',
                'attributionBold': true,
                'attributionUppercase': false,
                'attributionSpacingBelow': 36,
                'rolloverLinkUnderline': false,
                'rolloverLinkColor': '#00c800',
                'colorAdSeparator': '#202020'
            }
        };
    </script>
    <div id="main-wrap" class="clearfix">
        <div class="container clearfix">
            <div class="header clearfix">
                <div class="leftblk">
                    <!-- <img src="http://c.rmgserving.com/rmgpsc/7867/logo1.png" /> -->
                    <h3 class="domain_name">http.com</h3>
                </div>
                <div class="searchbox clearfix" id="searchbox"></div>
            </div>
            <div class="content clearfix">
                <div class="arrows frt_arr"></div>
                <div class="kwd_bloack clearfix" id="main_rs"></div>
                <div class="arrows lst_arr"></div>
            </div>
        </div>
        <div class="bottom_bg clearfix">
            <div class="bottom_rs clearfix">
                <!--<span class="separator"></span>-->

            </div>
        </div>
        <div class="footer-nav">
            <a href="#" onclick="return __gpup('http://www.http.com/rmgdsc/rprivacypolicy.php',600,800,'rpp');">Privacy Policy</a> </div>
    </div>
    <script type="text/javascript" language="javascript">
        window._debug = false;
        window._dumpdata = false;
        var _cflp = true;
        var pgld = true;
        var totRS = -1;
        var cntRS = 0;
        var _cfstc = false;
        var _cfblocker = true;
        var _cfrg = true;
        var _cfHelp = _cfHelp || {};
        _cfHelp.design = design || {};
        _cfHelp.servVars = {
            "webadfl": "webads.php",
            "_clhdl": "http:\/\/www.http.com\/rg-cltrack.php?&gr=%5B%7Crdr_url%7C%5D&gm=1WHI0lwrG0Sxkx4PsD8qyCOUd1dkaI2X1l8SGG1fMSd7bfeWMBF%2FyegsqR%2BZliS%2FkmMLg273bT8VsQ1oofKDtwu2BLulOOE9e9f%2FGbKVB3y9mwzTRLUiokCVqtb5Zk84C2EVcNaF2G3r9Bp3RsH1xfXrZC%2BE8YGoveyKL4ONQKcLbTlODzh%2BgHyFxmaFdMqH&gc=11101851242258318844884&gi=XYb9MaQozttVcq86LfylD8RNPuvlew8QArHjQzucy26g%2BlAY%2F2uHYh%2BA6ff50kcc1PxTBiLUWCU4jHCv3wduaWHfl%2Fk8mkRKtWu8EB0jglLzw7eeSqi3ynxhbG7655Fy%2FLKMke6nBiTt%2FuiCvqlrvF1W%2F0MGzePy07JWvEsPuPqNBujvQaw3gGby3VkjIxfm2atlkRoPbyOyKYholPIgVyyBcIBJQvM3kSCiRXOyaMpxjTprto%2B0LbFQmRT0%2Bv05GchdCqw0KwcBJr1GshmWISk5vXG28qmhuRBFtwmFZTdpEHObgOZuOhjtcRKzp6QcLHJxeH43ucmzlzBBC8rUhA%3D%3D",
            "kchst": "http:\/\/www.http.com\/",
            "kcpg": "render",
            "kcprm": "ga=5rKI1sSSBwFmmNY4RDXg%2BB7VS0yzNgnQ9JpHpsLHSsvFu0NyshpFxWu0exObFAo5IdFZQrTZujcZjrXONxp3PxtPV7ezER14zh7kWG69e5lf08niAgSfUKiyBSByBIQgw92X314XjQy5h09a65wi7JIhzrLV0PReUb%2F7SMvquCNLeg70qvVfr7kkYTx9Z4NO&gqsg=4OVo54jg6Fc8v4dgaRWvLlUoQfiT9l0g4KypdpyKIkp3XttitvahDJPyd8JV8AMP&maxads=0&gerf=NIVtBZ66fMb06L%2FzgSYcSUpDtBcVFLddp21rUKg8Zvs%3D&wait=true",
            "jtchdl": "0",
            "lkwtkn": null,
            "qry": null,
            "cntry": "IE",
            "prvid": 1566375,
            "lgky": "VFJIcUJEQUZtMGQ3dVY0amZaMmFIVi8vL2g2WUVmRkQxNVp6eEJRcSs1WFl3K3cvZFFISS9aN1Y4NE1KamNIUi95TW04Q3p6ek9RejR1N1pRemlJS1hORjIzNEF2Z3d3TGkyUGhGNkJxREdyRDJLNzZoWVMraXd0cFZGK1hiWTY5aGNYRTlmOHlDa3E5bmpLTHRtelJ3PT0%3D",
            "_aftrprss": "",
            "wclhdl": "http:\/\/www.http.com\/webclk?&gr=%5B%7Crdr_url%7C%5D&gm=N3DjGjxerDa%2B8Ln3%2B884CDaIWdGzXLcE6PGovrvm8cwEqFKj9%2FsSAw13RemOENdu59z53RIAX7qKedB%2BbC7k6lI6gFbOxQR7qkhujpZqsJMhUo0JnsgwBv6eY8R4QxT34RDgzTHG7pSdzqCdYGLMkjkrWeoByyf1I3drNmqLJoXF8CRTdNMoZwb5HxP8G2KL&gc=11101851242258318844501&gi=3asGzSvNXUezIegH6SkVMB9oHMTe%2BRtyB0XjQzqrg%2BegxXIIXBBR%2FNpmg%2FEpOC3SbbpbaUmlcSRdngRk3viiaUqqjkydC5SRcNAQIHAVCSz4CcvOsJgD4kJKepiaVN2MkCXSlbZfRbnV4Lj3QWLNCEpoUpJaD8GoKU2ZiKd%2B2wDKToaj236za8yEZZnAejv7CqvZi7bAnoT7LFHAT4YfVQYDvpwbpObRothY86AABfVVWr%2BDuudu2Z1HgnO7kRUJx945jHKd%2BHgj7szCxsb6xBb7wBlUblacTda5uwQKPLtiF5n7ChXCCJTf%2BYsC%2FDDegcmi71Z8FBC6SLSZ%2FYvsJQ%3D%3D",
            "is_webad_enabled": 0,
            "loaderimage": "http:\/\/c.rmgserving.com\/rmgisc\/loader.gif",
            "_afdad": 1,
            "noredir": 0,
            "erpub": "oversee11_3ph_adult_xml",
            "erch": "372",
            "erpubcln": "ca-dp-oversee11_3ph_xml",
            "erchcln": "000307",
            "ghu": "http:\/\/www.http.com\/rg-erdr.php?_rpo=t ntZzFFI&_rdm=9f1NJXf43BJvmfNf4.JvV&p=J1c4gcvmfN9ffFGoFgXoYVa%7C%40%7CbzGzTE%2C5f95zbT%2C5f95Et%7C%40%7CX55g.JvV%7C%40%7C%7C%40%7C%7C%40%7CZzbHzEZHE%7C%40%7CzHZZFTH%7C%40%7CK-%7C%40%7C19c4NW4cGFHzTEtTEGZGzTbT%7C%40%7Ct+nh8llbj%7C%40%7Ct+7zFZKFH&ga=5rKI1sSSBwFmmNY4RDXg%2BB7VS0yzNgnQ9JpHpsLHSsvFu0NyshpFxWu0exObFAo5IdFZQrTZujcZjrXONxp3PxtPV7ezER14zh7kWG69e5lf08niAgSfUKiyBSByBIQgw92X314XjQy5h09a65wi7JIhzrLV0PReUb%2F7SMvquCNLeg70qvVfr7kkYTx9Z4NO"
        };
        _cfHelp.newOpts = {
            "pageoptions": {
                "kw": "http",
                "pubId": "dp-oversee32_3ph_xml",
                "channel": "012174,test107,test49",
                "hl": "",
                "adtest": "off",
                "resultsPageBaseUrl": "http:\/\/www.http.com\/?ga=5rKI1sSSBwFmmNY4RDXg%2BB7VS0yzNgnQ9JpHpsLHSsvFu0NyshpFxWu0exObFAo5IdFZQrTZujcZjrXONxp3PxtPV7ezER14zh7kWG69e5lf08niAgSfUKiyBSByBIQgw92X314XjQy5h09a65wi7JIhzrLV0PReUb%2F7SMvquCNLeg70qvVfr7kkYTx9Z4NO&gqsg=4OVo54jg6Fc8v4dgaRWvLlUoQfiT9l0g4KypdpyKIkp3XttitvahDJPyd8JV8AMP&maxads=0&gerf=1NwSv6zgpkznlToCcC1LUPxAge73x0kzNUyoFwPpLQw%3D&wait=true",
                "clicktrackUrl": "http:\/\/www.http.com\/rg-cltrack.php?&gr=http%3A%2F%2Fgoogle.com%2F%3FRMGcaf%261437578625&gm=lCHzRGQVmF5GbI%2FoJNvHO9F0H1pK6b4imVg4sD83gr0zwhCg7kUY7A5JisFIua0ovWZYB4g4CbZlK6DDWhXlYS%2FoGwafEe2w6vLAAhtarOqSrags%2FXIb7w%2BHM0BEJgeyKmMXhVQya9xYRoVUupKo7SfWKOP26QW4BeaSRA7pvaleohDknqEJ8irgK41FOPBr&gc=11101851242258318844252&gi=dqJeJ2LIBm6EoVjrQxF%2FslKeVO2MMfAIGz5LwKTbfKTDn9L7haFS7W3vKNwbjSGhC5JthqkexETQM8mEun%2F4M5TCzM0Vphb237zGYUpoBtB1%2BdaJoSRcYUFKnr9lF6i32soKngUvevO7DHx9BHSz7sCMXYPLkJicrRb7WxfYNRWz0plNCCbrW8HYaPbhclR9gVbhSwIUdzpMjwGFrNyh%2B9iqRYF%2BX0wFzLunKbgTnLdmXJuP0Vm%2BjkU%2BC4rprcocV7pcHhlR0%2FKP2vHSKAlWTPcuCA9P04LbPqttMcdlol%2Fpm%2BAeIg79dx%2Bb28yVLKBEpMifhq6%2BT%2B2LndQdLFX%2FCoMvHjee%2BMfVwkRgY8%2F9hqFI%2B9UvBLgcgjCtTYSruCn7",
                "domainRegistrant": "as-drid-2351749742621707",
                "optimizeTerms": false,
                "terms": "[\"Domain Registration\",\"Network Monitoring\",\"Email Hosting\",\"Website Hosting\",\"Network Monitoring Software\",\"High Speed internet\",\"Cable Internet\",\"Broadband Internet\",\"Web Security\",\"Dial-Up Internet\",\"Wireless Internet Providers\",\"Unlimited Broadband\",\"Internet Speed Check\",\"Compare Broadband\",\"Cheap Internet Service\",\"Best Broadband Deals\",\"Internet Solutions\",\"Internet Music\",\"Internet Help\",\"Internet Telephony\",\"Business Internet Providers\",\"Broadband Availability\",\"Prepaid Wireless Internet\",\"Internet Florists\",\"Mobile Wireless Internet\"]",
                "uiOptimize": false,
                "domainName": "http.com"
            },
            "relatedsearch": {
                "adLoadedCallback": function(cnm, ld) {
                    if (pgld && !ld && _cfrg && !_cfstc && _cflp) {
                        cntRS++;
                        if (cntRS >= totRS && totRS != -1) {
                            window.location = "http://www.http.com/?_stc_=1"
                        }
                    }
                }
            },
            "textads": {
                "adLoadedCallback": function(cnm, ld) {}
            },
            "imagead": {
                "adLoadedCallback": function(cnm, ld) {}
            },
            "ads": {
                "adLoadedCallback": function(cnm, ld) {}
            }
        };
        _cfHelp.newOpts.pageoptions["pageLoadedCallback"] = function(requestAccepted, status) {
            this.onPageLoad(requestAccepted, status);
            _cfblocker = false
        };
        try {
            if (abp) {
                _cfHelp.newOpts.pageoptions['channel'] = _cfHelp.newOpts.pageoptions['channel'] + ',test101'
            }
        } catch (e) {}
    </script>
    <script type="text/javascript" src="http://a.rmgserving.com/rmgdsc/newcafv2.js?1.1"></script>
</body>
</html>
```

**Comment by stianeikeland** [comment]
Hello,
could you try without the http:// prefix?
```var etcd = new Etcd(['localhost:2379', 'localhost:4001']);```

I'm not sure where the html is coming from, but i would guess from http://http.com, as it's trying to access `http://http://localhost:2379` .. and your computer or dns-service probably adds .com to "help".

I might add a protocol detector, would make sense to support both `localhost:2379` and `http://localhost:2379` - avoids issues like this :)

As for the proxy warning. If no host:port is given it should default to `localhost:4001`. Try http://localhost:4001 in your browser, do you get the same proxy warning?

**Comment by jonniedarko** [comment]
So I tried without the prefix and still have the same issue. It is definitely a proxy issue as I have verified when I disable the proxy. but the  `NO_PROXY="127.0.0.1" node myprogram.js` does not work, I tried adding `export no_proxy="127.0.0.1, localhost"` to my bash_profile and still not working

**Comment by stianeikeland** [comment]
You could also try to unset the env-variable.
```
$ unset HTTP_PROXY
$ node program.js
```

**Comment by jonniedarko** [comment]
But there must be a way to do it without unsetting the proxy, if I need to connect to another host through the proxy in the same Node app I can't if I unset that proxy

**Comment by stianeikeland** [comment]
Yeah, you need to get it avoid using the proxy for local requests.
```
NO_PROXY="localhost,127.0.0.1"
# or
no_proxy="localhost,127.0.0.1"
```
seems to be the way to go based on googling, but if that doesn't work for you, then i'm a bit unsure how to proceed :/

I've confirmed that this works here:
```
# Following tries to use proxy that doesn't exist and fails:
HTTP_PROXY="localhost:9999" node program.js

# Following skips proxy and managed to connect to localhost.
HTTP_PROXY="localhost:9999" NO_PROXY="localhost,127.0.0.1" node program.js
```

**Comment by jonniedarko** [comment]
maybe its not working for me because I am using older version of node (v0.10.33) ?

**Comment by stianeikeland** [comment]
Could be, not sure, I tested setting NO_PROXY on node v0.12.7

**Comment by jonniedarko** [comment]
I have tried with the latest stable Node.js but still tries to go through the proxy.  I do not have any issue with any other framework. for example the request framework  on its own does not get this issue

**Comment by jonniedarko** [comment]
So I was able to get another module to work without turning off the proxy (nodejs-etcd])[https://www.npmjs.com/package/nodejs-etcd] , unfortunately its not as well maintained or documented as your module

**Comment by stianeikeland** [comment]
I've published a version under the beta tag, only thing it does is bump request library to latest version. There is a proxy related fix in the request changelog, I don't know, but it might help you up. Let me know if you try it and if it works :)

```
$ npm install node-etcd@beta
```

**Comment by jonniedarko** [comment]
@stianeikeland Thank you that has resolved the Issue :+1: is there anyway I can help to move this from beta to release?

**Comment by stianeikeland** [comment]
Released as v4.1.0 :)
